plugin/virustotal/vtreport.py

```diff
@@ -0,0 +1,404 @@
+      actions_dict['Crypto keys'] = sandbox.get('crypto_keys')
+    if sandbox.get('crypto_plain_text'):
+      actions_dict['Crypto plain texct'] = sandbox.get('crypto_plain_text')
+
+    if sandbox.get('text_decoded'):
+      actions_dict['Text decoded'] = sandbox.get('text_decoded')
+    if sandbox.get('text_highlighted'):
+      actions_dict['Text highlighted'] = sandbox.get('text_highlighted')
+
+    if sandbox.get('ja3_digests'):
+      actions_dict['JA3 digests'] = sandbox.get('ja3_digests')
+
+    if sandbox.get('modules_loaded'):
+      actions_dict['Modules loaded'] = sandbox.get('modules_loaded')
+
+    if sandbox.get('registry_keys_opened'):
+      actions_dict['Reg. keys opened'] = sandbox.get('registry_keys_opened')
+    if sandbox.get('registry_keys_set'):
+      # Returns a list of dictionaries
+      actions_dict['Reg. keys set'] = sandbox.get('registry_keys_set')
+    if sandbox.get('registry_keys_deleted'):
+      actions_dict['Reg. keys deleted'] = sandbox.get('registry_keys_deleted')
+
+    if sandbox.get('ids_alerts'):
+      # Returns a list of dictionaries
+      actions_dict['IDS alerts'] = sandbox.get('ids_alerts')
+
+    return actions_dict
+
+  def get_comments(self):
+    return self.comments
+
+  def get_contacted_ips(self):
+    return self.contacted_ips.keys()
+
+  def get_contacted_ip(self, ip):
+    """Returns a list containing ip + av detections ratio."""
+
+    ip_report = self.contacted_ips[ip]
+    ip_details = []
+    ip_analysis = ip_report.get('last_analysis_stats')
+    total = ip_analysis['harmless'] + ip_analysis['malicious']
+    total += ip_analysis['suspicious'] + ip_analysis['undetected']
+    ip_analysis = str(ip_analysis['malicious']) + '/' + str(total)
+    ip_details.append(ip)
+    ip_details.append(ip_analysis)
+    ip_details.append(ip_report.get('country'))
+    return ip_details
+
+  def get_embedded_ips(self):
+    return self.embedded_ips.keys()
+
+  def get_embedded_ip(self, ip):
+    """Return a list containing IPs + detections + country."""
+
+    ip_report = self.embedded_ips[ip]
+    ip_details = []
+    ip_analysis = ip_report.get('last_analysis_stats')
+    total = ip_analysis['harmless'] + ip_analysis['malicious']
+    total += ip_analysis['suspicious'] + ip_analysis['undetected']
+    ip_analysis = str(ip_analysis['malicious']) + '/' + str(total)
+    ip_details.append(ip)
+    ip_details.append(ip_analysis)
+    ip_details.append(ip_report.get('country'))
+    return ip_details
+
+  def get_contacted_urls(self):
+    return self.contacted_urls.keys()
+
+  def get_contacted_url(self, url):
+    """Return a list containing URL + detections."""
+
+    url_report = self.contacted_urls[url]
+    url_details = []
+    url_analysis = url_report.get('last_analysis_stats')
+    total = url_analysis['harmless'] + url_analysis['malicious']
+    total += url_analysis['suspicious'] + url_analysis['undetected']
+    url_analysis = str(url_analysis['malicious']) + '/' + str(total)
```

Comment:
Please use a = '%d/%d' % ( x, y) instead... it is much more readable.

Also please name it xxx_ratio so that is clear what this holds.